### PR TITLE
Performance enhancement to group lists

### LINKED ISSUE #505: Return last message in groups list
Add ability to sort by last message as well.

### PR DESCRIPTION
**Pull request by nplasterer**
Closes https://github.com/xmtp/xmtp-react-native/issues/505

This adds the ability to specify what group parameters you want loaded and passed on group listing since json encoding can be costly for large lists of members.
This also adds the ability to specify a sort order for the groups based on last message.

### CODE REVIEW

File: src/lib/types/GroupOptions.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export type GroupOptions = {
```
Comment (alexrisch) [inline src/lib/types/GroupOptions.ts:1]: Can we make sure these are getting exported from the index file?

File: src/index.ts [hunk 2]
```diff
@@ -1274,6 +1299,18 @@ interface AuthParams {
   historySyncUrl?: string
 }
 
+interface GroupParams {
```
Comment (alexrisch) [inline src/index.ts]: This looks like a duplicate of the GroupOptions in GroupOptions.ts

File: example/src/tests/groupPerformanceTests.ts [hunk 3]
```diff
@@ -268,3 +268,73 @@ test('testing sending message in large group', async () => {
 
   return true
 })
+
+test('testing large group listings with ordering', async () => {
```
Comment (alexrisch) [inline example/src/tests/groupPerformanceTests.ts]: 🔥

**Review by alexrisch — APPROVED** [review]

**Comment by nplasterer** [comment]
| Action  | Groups | iOS - Alix (Time ms) | iOS - Bo (Time ms) | Android - Alix (Time ms) | Android - Bo (Time ms) |
|---------|--------|----------------------|--------------------|--------------------------|------------------------|
| Loaded  | 1000   | 1725                 | 1516               | 3686                     | 3733                   |
| Loaded w params  | 1000   | 832                  | 424                | 2632                     |       1671                 |
| Synced  | 1000   | 11                   | 4784               | 16                       | 12755                  |

**Pull request merged**